ivy/functional/backends/jax/experimental/layers.py

```diff
@@ -235,9 +237,9 @@ def max_pool2d(
     )
 
     if data_format == "NCHW":
-        res = jnp.transpose(res, (0, 3, 1, 2))
+        return jnp.transpose(res, (0, 3, 1, 2))
 
-    return res.astype(odtype)
+    return res
 
```

Comment:
Okay last thing, are these changes to `max_pool2d` related to the PR? If not you should make a separate commit.

Reply:
No, will revert this to the original logic